### PR TITLE
feat: Add Jmeter check

### PR DESCRIPTION
**Pull request by Kaitou786**
Signed-off-by: Tarun Khandelwal <tarun@flanksource.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -583,6 +595,42 @@ This Check will
 
 
 
+### Jmeter - Run the supplied JMX test plan against the specified host
+
+This check will execute the jmeter cli to execute the JMX test plan on the specified host
+
+When running Jmeter check one can increase the memory limits due to the usage of JRE inside the canary-checker pod 
+
+Note: The default memory limit is set to 512Mi
```
Comment (moshloop) [inline README.md:604]:
```suggestion
> Note that JMeter is a memory hungery java application and you will likely need to increase the default memory limit from 512Mi to 1 -2Gi or higher depending on the complexity, count and frequency of jmeter tests
```

File: api/v1/checks.go [hunk 2]
```diff
@@ -196,6 +196,44 @@ func (c ResticCheck) GetType() string {
 	return "restic"
 }
 
+type JmeterCheck struct {
+	// JmxFrom defines tge ConfigMap or Secret reference to get the JMX test plan
+	JmxFrom kommons.EnvVar `yaml:"jmxFrom" json:"jmxFrom"`
```
Comment (moshloop) [inline api/v1/checks.go:201]:
```suggestion
	Jmx kommons.EnvVar `yaml:"jmxFrom" json:"jmx"`
```

If EnvVar has the valueFrom then it shouldn't be duplicted in the var name

File: checks/jmeter.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	var host string
+	var port string
+	if jmeterCheck.Host != "" {
+		host = "-H " + jmeterCheck.Host
+	}
+	if jmeterCheck.Port != 0 {
+		port = "-P " + string(jmeterCheck.Port)
+	}
+	jmeterCmd := fmt.Sprintf("jmeter -n %s %s -t %s %s %s", getProperties(jmeterCheck.Properties), getSystemProperties(jmeterCheck.SystemProperties), filename, host, port)
+	err = exec.Exec(jmeterCmd)
```
Comment (moshloop) [inline checks/jmeter.go:65]: I think we may need to parse the jmeter results and check if the tests actually passed, and then also compare the duration as reported by jmeter (i.e. excluding JRE and jmeter start times) against a responseDuration and content and bubbling up any errors jmeter reports up

File: test/setup.yml [hunk 4]
```diff
@@ -292,3 +292,91 @@ metadata:
 type: Opaque
 data:
   value: Zm9v
+
+---
+apiVersion: v1
+data:
+  sampleTest.jmx: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<jmeterTestPlan version=\"1.2\"
```
Comment (moshloop) [inline test/setup.yml:299]: You can use a multiline yaml string to format this better

File: api/v1/checks.go (hunk 2 above)
Comment (Kaitou786) [reply api/v1/checks.go]: ack

File: checks/jmeter.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	jmeterCmd := fmt.Sprintf("jmeter -n %s %s -t %s %s %s -l %s", getProperties(jmeterCheck.Properties), getSystemProperties(jmeterCheck.SystemProperties), testPlanFilename, host, port, logFilename)
+	_, ok := exec.SafeExec(jmeterCmd)
+	if !ok {
+		return Failf(jmeterCheck, "error running the jmeter command: %v", jmeterCmd)
+	}
+	f, err := os.Open(logFilename)
+	if err != nil {
+		return Failf(jmeterCheck, "error opening the log file: %v", err)
+	}
+	defer f.Close()
```
Comment (moshloop) [inline checks/jmeter.go:77]: These defers should move to straight after resource creation

**Pull request merged**